src/nix/hash.cc

```diff
@@ -99,15 +65,78 @@ struct CmdHashBase : Command
     }
 };
 
+/**
+ * `nix hash path`
+ */
+struct CmdHashPath : CmdHashBase
+{
+    CmdHashPath()
+        : CmdHashBase(FileIngestionMethod::Recursive)
+    {
+        addFlag(flag::hashAlgo("algo", &hashAlgo));
+        addFlag(flag::fileIngestionMethod(&mode));
+        addFlag(flag::hashFormatWithDefault("format", &hashFormat));
+        #if 0
+        addFlag({
+            .longName = "modulo",
+            .description = "Compute the hash modulo the specified string.",
+            .labels = {"modulus"},
+            .handler = {&modulus},
+        });
+        #endif
+    }
+};
+
+/**
+ * For deprecated `nix hash file`
+ *
+ * Deprecation Issue: https://github.com/NixOS/nix/issues/8876
+ */
+struct CmdHashFile : CmdHashBase
+{
+    CmdHashFile()
+        : CmdHashBase(FileIngestionMethod::Flat)
+    {
+        addFlag({
+            .longName = "sri",
+            .description = "Print the hash in SRI format.",
+            .handler = {&hashFormat, HashFormat::SRI},
+        });
+
+        addFlag({
+            .longName = "base64",
+            .description = "Print the hash in base-64 format.",
+            .handler = {&hashFormat, HashFormat::Base64},
+        });
+
+        addFlag({
+            .longName = "base32",
+            .description = "Print the hash in base-32 (Nix-specific) format.",
+            .handler = {&hashFormat, HashFormat::Nix32},
+        });
+
+        addFlag({
+            .longName = "base16",
+            .description = "Print the hash in base-16 format.",
+            .handler = {&hashFormat, HashFormat::Base16},
+        });
+
+        addFlag(flag::hashAlgo("type", &hashAlgo));
```

Comment:
Deprecated Flags for deprecated `nix hash file`; all these are subsumed by the new ones on `nix hash path`.

Reply:
If they are deprecated, I think they should be marked as such

Reply:
I'm not clear why these are kept for `nix hash file` and not `nix hash path`. Both commands accept them similarly right now, so we should either drop them for both, or (preferably) keep them for both. But keeping them for only one seems weird and inconsistent. Or am I missing something?

Reply:
`nix hash file` is a deprecated command, `nix hash path` is not.

As described in the issue, the only non-deprecated new CLI commands are
 - `nix hash path`
 - `nix hash convert`
- `nix hash path`

Reply:
I didn't deprecate the flags because the whole command they are on should be deprecated. I meant "these flags are just for the deprecated command, so I'll move them there". Sorry that first sentence was confusing.

Reply:
> `nix hash file` is a deprecated command, `nix hash path` is not.

Ah right, I had forgotten about that distinction.

At any rate, I don't think we should remove them, just mark them as deprecated

Reply:
We don't have a mechanism for that, unfortunately. But I put them back on `nix hash path` (in addition to `nix hash file`) and added a `FIXME`.